tests/integration/conftest.py

```diff
@@ -51,15 +54,37 @@ def pytest_collection_modifyitems(config, items):
 
 
 @pytest.fixture(scope="session", autouse=True)
-def logging_setup():
+def test_env_setup():
+    _logging_setup()
+    old_state = libnmstate.show()
+    _empty_net_state()
+    _ethx_init()
+    yield
+    libnmstate.apply(old_state, verify_change=False)
```

Comment:
If this operation fails/raises, no one will call the following one.

Reply:
+1 here, I missed this. Sorry.